upstairs/src/lib.rs

```diff
@@ -4438,14 +4379,36 @@ impl Downstairs {
                 } => {
                     /*
                      * For a read, make sure the data from a previous read
-                     * has the same hash
+                     * has the same list of hashes
                      */
                     let read_data: Vec<ReadResponse> = read_data.unwrap();
                     assert!(!read_data.is_empty());
-                    if job.read_response_hashes != read_response_hashes {
+
+                    let mismatch = std::iter::zip(
+                        job.read_response_hashes.iter(),
+                        read_data.iter().map(|response| {
+                            response.block_contexts.get(0).map(|ctx| ctx.hash)
```

Comment:
This is used enough that it could be a function in `ReadResponse`, e.g. `first_hash(&self) -> Option<u64>`

Reply:
agreed, done in 6eb6d22